core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -399,23 +429,28 @@ private void configureConnectors(SslContextFactory sslContextFactory) {
                               && config.getBoolean(RestConfig.HTTP2_ENABLED_CONFIG);
 
     @SuppressWarnings("deprecation")
-    List<URI> listeners = parseListeners(config.getList(RestConfig.LISTENERS_CONFIG),
-            config.getInt(RestConfig.PORT_CONFIG), Arrays.asList("http", "https"), "http");
+    List<NamedURI> listeners = parseListeners(
+        config.getList(RestConfig.LISTENERS_CONFIG),
+        config.getListenerProtocolMap(),
+        config.getInt(RestConfig.PORT_CONFIG),
+        SUPPORTED_URI_SCHEMES, "http");
 
-    for (URI listener : listeners) {
+    for (NamedURI listener : listeners) {
       addConnectorForListener(httpConfiguration, httpConnectionFactory, listener, http2Enabled);
     }
   }
 
   private void addConnectorForListener(HttpConfiguration httpConfiguration,
                                        HttpConnectionFactory httpConnectionFactory,
-                                       URI listener,
+                                       NamedURI listener,
                                        boolean http2Enabled) {
     NetworkTrafficServerConnector connector;
 
     if (http2Enabled) {
-      log.info("Adding listener with HTTP/2: " + listener.toString());
-      if (listener.getScheme().equals("http")) {
+      log.info(
+          "Adding listener " + (listener.name == null ? "" : "'" + listener.name + "'")
```

Comment:
This should be in the NameURI toString().

Reply:
well pointed out